packages/react-input/src/components/Input/useInput.ts

```diff
@@ -29,12 +30,26 @@ export const useInput = (props: InputProps, ref: React.Ref<HTMLElement>): InputS
       contentBefore: 'span',
       contentAfter: 'span',
     },
-    input: resolveShorthand(input, { required: true }),
-    contentAfter: resolveShorthand(contentAfter),
-    contentBefore: resolveShorthand(contentBefore),
-    root: getNativeElementProps('span', {
-      ref,
-      ...props,
+    input: resolveShorthand(props.input, {
+      required: true,
+      defaultProps: {
+        type: 'text',
+        ref,
+        ...nativeProps.primary,
+      },
+    }),
+    contentAfter: resolveShorthand(props.contentAfter),
+    contentBefore: resolveShorthand(props.contentBefore),
+    root: resolveShorthand(props.root, {
+      required: true,
+      defaultProps: nativeProps.root,
     }),
   };
+
+  state.input.onChange = useEventCallback(ev => {
```

Comment:
Similar to my comment on your Checkbox PR: https://github.com/microsoft/fluentui/pull/20901#discussion_r761433958, this will override the `onChange` handler if it is provided on the `input` slot.

Reply:
I think this is no longer relevant if we merge #21007?